docs/DeveloperGuide.md

```diff
@@ -249,47 +249,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+2. TARS save the task and add the command to command history<br>
+Use case ends.
+
+**Extensions**
+
+2a. The format is invalid
+
+> 2a1. TARS shows an error message<br>
+  Use case resumes at step 1
+
+2b. The end datetime is smaller than start datetime
+
+> 2b1. TARS shows an error message<br>
+  Use case resumes at step 1
+  
+#### Use case: UC03- Delete task
+
+**MSS**
+
+1. User requests to list tasks
+2. TARS shows a list of tasks
+3. User requets to delete a specific task in the list
+4. TARS deletes the task<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> 2a1. Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+#### Use case: UC04 - Edit task
+
+**MSS**
+
+1. User requests to list tasks
+2. TARS shows a list of tasks
+3. User requests to edit a specific task in the list
+4. TARS updates the task<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> 2a1. Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+3b. The format is invalid
+
+> 3b1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+#### Use case: UC05 - Edit task by appending details
+
+**MSS**
+
+1. User requests to list tasks
+2. TARS shows a list of tasks
+3. User requests to edit a specific task in the list by appending details
+4. TARS updates the task<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> 2a1. Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+3b. The format is invalid
+
+> 3b1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+#### Use case: UC06 - Edit tag name
+  
+**MSS**
+
+1. User requests to list tags
+2. TARS shows a list of tags
+3. User requests to edit a specific tag in the list
+4. TARS updates the tag<br>
 Use case ends.
 
 **Extensions**
 
 2a. The list is empty
 
-> Use case ends
+> 2a1. Use case ends
 
 3a. The given index is invalid
 
-> 3a1. AddressBook shows an error message <br>
+> 3a1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+3b. The format is invalid
+
+> 3b1. TARS shows an error message<br>
   Use case resumes at step 2
+  
+#### Use case: UC07 - Delete tag
+
+**MSS**
+
+1. User requests to list tags
+2. TARS shows a list of tags
+3. User requets to delete a specific tag in the list
+4. TARS deletes the tag<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> 2a1. Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+#### Use case: UC08 - List tags
+
+**MSS**
+
+1. User requests to list tags
+2. TARS shows a list of tags<br>
+Use case ends.
+
+#### Use case: UC09 - Undo a previous command
+
+**MSS**
+
+1. User requests to undo a previous command
+2. TARS unexecutes (undo) the last command in the history list<br>
```

Comment:
Hi team, I don't think you can unexecute something, I think you meant a simple reinstating the task's status etc right ~ just like if you drink water you can spit it out but you can't un drink it XD

It is ok to express in simple words as long as they are clear :)